core/services/registrysyncer/local_registry.go

```diff
@@ -0,0 +1,81 @@
+package registrysyncer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+type CapabilityID string
+type DonID uint32
+
+type DON struct {
+	capabilities.DON
+	CapabilityConfigurations map[CapabilityID]capabilities.CapabilityConfiguration
+}
+
+type Capability struct {
+	ID             CapabilityID
+	CapabilityType capabilities.CapabilityType
+}
+
+type LocalRegistry struct {
+	lggr              logger.Logger
+	peerWrapper       p2ptypes.PeerWrapper
+	IDsToDONs         map[DonID]DON
+	IDsToNodes        map[p2ptypes.PeerID]kcr.CapabilitiesRegistryNodeInfo
+	IDsToCapabilities map[CapabilityID]Capability
+}
+
+func (l *LocalRegistry) GetLocalNode(ctx context.Context) (capabilities.Node, error) {
+	if l.peerWrapper.GetPeer() == nil {
```

Comment:
what is the peerWrapper representing? 

is this call supposed to be a lookup about something scoped to the the the local registry itself? 

please add a comment about the intention; it reads both as a call to some neighbor and as a loading something about itself